api/src/pcapi/core/users/generator.py

```diff
@@ -10,15 +11,28 @@
 @dataclasses.dataclass
 class GenerateUserData:
     age: int
-    is_email_validated: bool
     is_beneficiary: bool
+    step: subscription_models.SubscriptionStep | None = None
 
 
 def generate_user(user_data: GenerateUserData) -> users_models.User:
     if not settings.ENABLE_TEST_USER_GENERATION:
         raise exceptions.UserGenerationForbiddenException("Test user generation is disabled")
 
+    factory = users_factories.BaseUserFactory
+    match user_data.step:
+        case subscription_models.SubscriptionStep.EMAIL_VALIDATION:
+            factory = users_factories.EmailValidatedUserFactory
+        case subscription_models.SubscriptionStep.PHONE_VALIDATION:
+            factory = users_factories.PhoneValidatedUserFactory
+        case subscription_models.SubscriptionStep.PROFILE_COMPLETION:
+            factory = users_factories.ProfileCompletedUserFactory
+        case subscription_models.SubscriptionStep.IDENTITY_CHECK:
+            factory = users_factories.IdentityValidatedUserFactory
+        case subscription_models.SubscriptionStep.HONOR_STATEMENT:
+            factory = users_factories.HonorStatementValidatedUserFactory
+
     if user_data.is_beneficiary:
-        return users_factories.BeneficiaryFactory(age=user_data.age)
+        factory = users_factories.BeneficiaryFactory
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Si `BENEFICIARY` devient la dernière step, on pourra rentrer ce if dans le switch au-dessus
Je suis de plus en plus convaincu par ça !
Dans une autre PR par contre, ici c'est :+1: